File: DuDoong-Api/src/main/java/band/gosrock/api/common/aop/hostPartner/HostPartnerCallTransactionFactory.java
package band.gosrock.api.common.aop.hostPartner;


import band.gosrock.api.common.aop.hostRole.FindHostFrom;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Component;

@RequiredArgsConstructor
@Component
class HostPartnerCallTransactionFactory {

    private final HostPartnerEventTransaction hostRoleEventTransaction;
    private final HostPartnerHostTransaction hostRoleHostTransaction;

    public HostPartnerCallTransaction getCallTransaction(FindHostFrom findHostFrom) {
        if (findHostFrom == FindHostFrom.HOST_ID) {
            return hostRoleHostTransaction;
        }
        return hostRoleEventTransaction;
    }
}

File: DuDoong-Api/src/main/kotlin/band/gosrock/api/common/aop/hostPartner/HostPartnerCallTransactionFactory.kt
package band.gosrock.api.common.aop.hostPartner

import band.gosrock.api.common.aop.hostRole.FindHostFrom
import org.springframework.stereotype.Component

@Component
internal class HostPartnerCallTransactionFactory(
    private val hostRoleEventTransaction: HostPartnerEventTransaction,
    private val hostRoleHostTransaction: HostPartnerHostTransaction
) {
    fun getCallTransaction(findHostFrom: FindHostFrom): HostPartnerCallTransaction {
        return if (findHostFrom == FindHostFrom.HOST_ID) {
            hostRoleHostTransaction
        } else {
            hostRoleEventTransaction
        }
    }
}
